pilot/pkg/model/service.go

```diff
@@ -1261,22 +1261,26 @@ func (s *Service) GetAddressForProxy(node *Proxy) string {
 // GetExtraAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
 // Especially for dual stack k8s service to get other IP family addresses.
 func (s *Service) GetExtraAddressesForProxy(node *Proxy) []string {
-	if features.EnableDualStack && node.Metadata != nil {
-		if node.Metadata.ClusterID != "" {
-			addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
-			if len(addresses) > 1 {
-				return addresses[1:]
-			}
-		}
+	addresses := s.getAllAddressesForProxy(node)
+	if len(addresses) > 1 {
+		return addresses[1:]
 	}
 	return nil
 }
 
 // GetAllAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
 // Especially for dual stack k8s service to get other IP family addresses.
 func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
-	if (features.EnableDualStack || features.EnableAmbient) && node.Metadata != nil && node.Metadata.ClusterID != "" {
+	return s.getAllAddressesForProxy(node)
+}
+
+func (s *Service) getAllAddressesForProxy(node *Proxy) []string {
+	if node.Metadata != nil && node.Metadata.ClusterID != "" {
 		addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
+		if (features.EnableDualStack || features.EnableAmbient) && len(addresses) > 0 {
```

Comment:
donot we need to check whether pod support dual stack?

Reply:
Like a multi cluster mesh, one cluster support only ipv4, the other support dual stack

Reply:
No, this function returns addresses relevant for the cluster of the requesting proxy.

Btw. this is irrelevant in this PR, as I didn't change anything related to dual-stack.

Reply:
Yes, none of your bussiness. Just see this and indeed it is a bug

Reply:
In general, we shouldn't be making decisions based on `EnableDualStack` and should treat it as a deprecated field - it's almost never the correct signal to check.